### PR TITLE
Fix clog2text not being available during test runs

### PR DESCRIPTION
**Pull request by thhous-msft**
Now generates clog2text as a build step, and then loads that build step onto the runner.

### CODE REVIEW

File: scripts/log.ps1 [hunk 1]
```diff
@@ -200,10 +200,16 @@ function Log-Stop {
...
-            # $Command = "$Clog2Text_lttng -i $BableTraceFile -s $SideCar -o $ClogOutputDecodeFile --showTimestamp --showCpuInfo"
-            # Write-Host $Command
-            # Invoke-Expression $Command | Write-Debug
+            try {
+                Write-Host "Decoding into human-readable text: $ClogOutputDecodeFile"
+                $Command = "$Clog2Text_lttng -i $BableTraceFile -s $SideCar -o $ClogOutputDecodeFile --showTimestamp --showCpuInfo"
+                Write-Host $Command
+                Invoke-Expression $Command | Write-Debug
+            } catch {
+
```
Comment (anrossi) [inline scripts/log.ps1:209]: maybe log a statement when this fails?

File: scripts/log.ps1 [hunk 2]
```diff
@@ -242,10 +248,14 @@ function Log-Decode {
...
-        #$Command = "$Clog2Text_lttng -i $BableTraceFile -s $SideCar -o $ClogOutputDecodeFile"
-        #Write-Host $Command
-        #Invoke-Expression $Command
+        try {
+            Write-Host "Decoding Babeltrace into human text using CLOG"
+            $Command = "$Clog2Text_lttng -i $BableTraceFile -s $SideCar -o $ClogOutputDecodeFile"
+            Write-Host $Command
+            Invoke-Expression $Command
+        } catch {
+
```
Comment (anrossi) [inline scripts/log.ps1:257]: Same here

**Review by anrossi — APPROVED** [review]

File: scripts/prepare-machine.ps1 [hunk 3]
```diff
@@ -438,8 +443,31 @@ if ($InstallJOM) { Install-JOM }
 if ($InstallCodeCoverage) { Install-OpenCppCoverage }
 if ($InstallSigningCertificate) { Install-SigningCertificate }
 if ($InstallTestCertificates) { Install-TestCertificates }
+function Install-DotnetTool {
```
Comment (nibanks) [inline scripts/prepare-machine.ps1:446]: Please put this function up with all the rest.

File: scripts/log.ps1 [hunk 4]
```diff
@@ -196,15 +196,19 @@ function Log-Stop {
...
-            # Remove-Item -Path $BableTraceFile -Force | Out-Null
+            try {
+                Write-Host "Decoding into human-readable text: $ClogOutputDecodeFile"
+                $Command = "$Clog2Text_lttng -i $BableTraceFile -s $SideCar -o $ClogOutputDecodeFile --showTimestamp --showCpuInfo"
+                Write-Host $Command
+                Invoke-Expression $Command | Write-Debug
+            } catch {
+                $err = $_
+                Write-Host "Failed to decode logs."
+                Write-Host "Babeltrace ran. Compile clog2text and run the following command"
```
[the remark this answers is not in the document]
Comment (nibanks) [reply scripts/log.ps1:207]: Or at least tell them the exact commands to run.

[the remark this answers is not in the document]
Comment (thhous-msft) [reply scripts/log.ps1:207]: This does tell them the exact command to run. "Compile clog2text and run the following command" And then prints the command.

[the remark this answers is not in the document]
Comment (nibanks) [reply scripts/log.ps1:207]: I mean the command to compile it.

**Review by nibanks — DISMISSED** [review]

**Review by nibanks — APPROVED** [review]

**Pull request merged**